### ISSUE TITLE
Number of steps

### ISSUE THREAD

**Issue by rogerguess** [tensorflow/models#1611]
-https://github.com/tensorflow/models/tree/master/attention_ocr
-No custom code, following instructions
-Ubuntu 14.04.5 
-TensorFlow installed from source
-('v1.1.0-rc0-61-g1ec6ed5', '1.1.0')
-following steps from instructions: https://github.com/tensorflow/models/tree/master/attention_ocr#how-to-use-this-code for  checkpoint_inception=inception_v3.ckpt

I am running train.py but am unable to see totals steps in the checkpoint or steps remaining in order to calculate time to complete training. Is there a way to see the total steps remaining and add that to the training output?

**Comment by aselle** [comment]
@nealwu, this would probably be a great feature to have on all garden models, perhaps as a library in tensorflow that could be used by even non-tensor garden models.

**Comment by nealwu** [comment]
Thanks Andy, we'll keep this in mind for the garden -- `tf.estimator` essentially provides this because we pass in a maximum number of training steps when running the model. In the meantime, I'll reassign this to @alexgorban who maintains the particular model mentioned.

**Comment by nealwu** [comment]
Looks like I can't officially reassign to him to, so @alexgorban can you take a look?

**Comment by rogerguess** [comment]
Thank you for considering this feature. 

Is there a way today that I can tell how many steps are in checkpoint_inception=inception_v3.ckpt?

**Comment by nealwu** [comment]
@rogerguess looks like Alex is out this week unfortunately but he'll be able to take a closer look next week.

**Comment by rogerguess** [comment]
no problem, thank you.

**Comment by alexgorban** [comment]
> I am running train.py but am unable to see totals steps in the checkpoint or steps remaining in order to calculate time to complete training. Is there a way to see the total steps remaining and add that to the training output?

the `train.py` script will do the number of steps specified as `--max_number_of_steps` flag (1e10 by default), regardless of number of steps "in the checkpoint". The training script outputs the current step number. In order to calculate time to complete a training you need to watch the loss and full sequence accuracy graphs in the TensorBoard to see/decide when the model converged. The reported full sequence accuracy was reached after 400K steps. So the benefits of showing a n/max_number_of_steps progress bar is not clear to me.

Anyway if you need to show how many steps were performed before a checkpoint was saved you can use the `tensorflow/python/tools/inspect_checkpoint.py`:
```
python inspect_checkpoint.py --file_name=<filename> --tensor_name=global_step
```

but unfortunately the inception_v3.ckpt doesn't have this information and for the model.ckpt-399731 it shows the number which is also obvious from the filename itself:
```
inspect_checkpoint.py --file_name=model.ckpt-399731 --tensor_name=global_step
tensor_name:  global_step
399732
```

**Comment by rogerguess** [comment]
@alexgorban , thank you for the explanation and see what you mean about this not being helpful/necessary. I am new to all this, and did not realize the checkpoint filename actually told you the number in the filename. I was asking because I was trying to determine how far along my training was or had remaining. The approach I was using (ubuntu virtual machine) was far too slow and I have purchased a GPU to make things faster. Thank you.
